subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -2815,6 +2815,75 @@ static bool json_item_string_exists(const cJSON *const obj, const char *const ke
 	return (strcmp(str_val, val) == 0);
 }
 
+static int nrf_cloud_parse_location_anchors_json(const cJSON * const loc_obj,
+	struct nrf_cloud_location_result * const location_out)
+{
+	cJSON *anchors, *mac, *name;
+	size_t buf_idx = 0;
+	size_t node_sz;
+	size_t name_sz;
+	int ret = 0;
+
+	/* Init anchor output */
+	sys_slist_init(&location_out->anchor_list);
+	if (location_out->anchor_buf_sz && location_out->anchor_buf) {
+		location_out->anchor_buf[0] = '\0';
+	}
+
+	/* Anchor info is provided in an array of objects */
+	anchors	= cJSON_GetObjectItem(loc_obj, NRF_CLOUD_LOCATION_JSON_KEY_ANCHORS);
+	location_out->anchor_cnt = (uint32_t)cJSON_GetArraySize(anchors);
+
+	for (int idx = 0; idx < location_out->anchor_cnt; ++idx) {
+		struct nrf_cloud_anchor_list_node *node;
+		cJSON *anc = cJSON_GetArrayItem(anchors, idx);
+
+		mac	= cJSON_GetObjectItem(anc, NRF_CLOUD_LOCATION_JSON_KEY_ANC_MAC);
+		name	= cJSON_GetObjectItem(anc, NRF_CLOUD_LOCATION_JSON_KEY_ANC_NAME);
+
+		if (cJSON_IsString(mac)) {
+			LOG_DBG("Wi-Fi anchor MAC: %s", mac->valuestring);
+		}
+
+		if (cJSON_IsString(name)) {
+			LOG_DBG("Wi-Fi anchor name: %s", name->valuestring);
+		} else {
+			continue;
+		}
+
+		if (!(location_out->anchor_buf_sz && location_out->anchor_buf)) {
+			/* No anchor buffer provided */
+			continue;
+		}
+
+		/* Get the size of the anchor name and node container */
+		name_sz = strlen(name->valuestring) + 1;
+		node_sz = sizeof(*node) + name_sz;
+
+		/* Ensure node will fit in buffer */
+		if ((buf_idx + node_sz) > location_out->anchor_buf_sz) {
+			LOG_WRN("Anchor does not fit in provided buffer");
+			/* Reset the list and buffer */
+			sys_slist_init(&location_out->anchor_list);
```

Comment:
Instead of resetting the list, could we keep the list and update `anchor_cnt`?

Reply:
i'd prefer to keep `anchor_cnt` unchanged.
how about keeping the list data and updating the comments?

```
	/** List of received anchor names contained as @ref nrf_cloud_anchor_list_node.
	 *  If all anchor names cannot fit in the list, the number of items in the list
	 *  will be less than @ref anchor_cnt.
	 */
	sys_slist_t anchor_list;
```

Reply:
Perfect. This avoids any confusion, I like it.